### PR TITLE
Implement auto start on login

### LINKED ISSUE #288: Implement auto start on login.
**Branch:** `develop` (os x installers)

**Expected behavior:** `KA Lite` application must  auto start if the user check the "Auto start on login" checkbox in KA Lite preferences dialog.

### PR DESCRIPTION
**Pull request by amodia**
@cpauya this fixes #288

### CODE REVIEW

File: osx/KA-Lite/KA-Lite/AppDelegate.m [hunk 1]
```diff
@@ -712,30 +722,32 @@ BOOL setEnvVars() {
...
+    if ([[NSUserDefaults standardUserDefaults] boolForKey:@"autoStartOnLogin"]){
+        NSString *kaliteAppPath = [NSString stringWithFormat:@"%@%@", [[NSBundle mainBundle] bundlePath], @"/Contents/MacOS/KA-Lite"];
+         launchStr = [NSString stringWithFormat:@"%@ ; %@", KaliteHomeStr, kaliteAppPath];
+    }
+   
     NSArray *arr = @[@"sh", @"-c", launchStr];
     [plistDict setObject:arr forKey:@"ProgramArguments"];
     [plistDict setObject:[NSNumber numberWithBool:TRUE] forKey:@"RunAtLoad"];
-    showNotification([NSString stringWithFormat:@"Setting KALITE_HOME and KALITE_PYTHON environment variables... %@", plistDict]);
+    showNotification([NSString stringWithFormat:@"Setting KALITE_HOME environment variables... %@", plistDict]);
```
Comment (cpauya) [inline osx/KA-Lite/KA-Lite/AppDelegate.m]: uhm, variables?

File: osx/KA-Lite/KA-Lite/AppDelegate.m [hunk 2]
```diff
@@ -712,30 +722,32 @@ BOOL setEnvVars() {
         return FALSE;
     }
     
-    
     // Path of the KALITE_PYTHON environment variable
```
Comment (cpauya) [inline osx/KA-Lite/KA-Lite/AppDelegate.m]: Are these still needed?

Comment (cpauya) [inline osx/KA-Lite/KA-Lite/AppDelegate.m]: I mean we should check if `KALITE_PYTHON` is valid when the app starts, then prompt the user.  

Making the check here in `setEnvVars()`, and returning `FALSE` if it's invalid, doesn't make sense.

File: osx/KA-Lite/KA-Lite/AppDelegate.m [hunk 3]
```diff
@@ -712,30 +722,32 @@ BOOL setEnvVars() {
...
     NSMutableDictionary *plistDict = [[NSMutableDictionary alloc] init];
     [plistDict setObject:org forKey:@"Label"];
     
-    
-    NSString *launchStr = [NSString stringWithFormat:@"%@ ; %@", KalitePythonStr, KaliteHomeStr];
+    // Append KA Lite app path if autoStartOnLogin value is TRUE.
+    NSString *launchStr = [NSString stringWithFormat:@"%@", KaliteHomeStr];
+    if ([[NSUserDefaults standardUserDefaults] boolForKey:@"autoStartOnLogin"]){
+        NSString *kaliteAppPath = [NSString stringWithFormat:@"%@%@", [[NSBundle mainBundle] bundlePath], @"/Contents/MacOS/KA-Lite"];
+         launchStr = [NSString stringWithFormat:@"%@ ; %@", KaliteHomeStr, kaliteAppPath];
```
Comment (cpauya) [inline osx/KA-Lite/KA-Lite/AppDelegate.m]: @amodia Is this the code responsible for creating the .plist so that app will auto-start on login?  

Fix the indention here please.

File: osx/KA-Lite/KA-Lite/AppDelegate.m [hunk 4]
```diff
@@ -712,30 +722,32 @@ BOOL setEnvVars() {
...
+    }
+   
     NSArray *arr = @[@"sh", @"-c", launchStr];
     [plistDict setObject:arr forKey:@"ProgramArguments"];
     [plistDict setObject:[NSNumber numberWithBool:TRUE] forKey:@"RunAtLoad"];
-    showNotification([NSString stringWithFormat:@"Setting KALITE_HOME and KALITE_PYTHON environment variables... %@", plistDict]);
+    showNotification([NSString stringWithFormat:@"Setting KALITE_HOME environment variables... %@", plistDict]);
     
     // Override org.learningequality.kalite.plist content
     BOOL ret = [plistDict writeToFile:target atomically:YES];
```
Comment (cpauya) [inline osx/KA-Lite/KA-Lite/AppDelegate.m]: Can we have an exception handler here?  Unless the `writeToFile` method handles it already and will always return a `BOOL`?

**Comment by cpauya** [comment]
@amodia Made some suggestions, looking good so far!  Hope @djallado can test it too?

**Comment by amodia** [comment]
@cpauya this PR is still work in progress. I will notify you if it's done.

**Comment by amodia** [comment]
@djallado please test this PR.

**Comment by djallado** [comment]
@amodia, I tested your PR, and the auto login is working on me. I have no issue when I click on `Start KA Lite` in menu option during first load.

File: osx/post_installation.sh [hunk 5]
```diff
@@ -113,12 +113,12 @@ update_env
...
 
 ((STEP++))
-echo "$STEP/$STEPS. Create plist in ~/Library/LaunchAgents folders..."
-if [ ! -d "$LAUNCH_AGENTS" ]; then
-    echo ".. Must create '$LAUNCH_AGENTS' folder..."
-    sudo mkdir -p $LAUNCH_AGENTS
+echo "$STEP/$STEPS. Remove plist in /Library/LaunchAgents folder..."
+if [ ! -f "$PLIST_SRC" ]; then
+    echo ".. Must remove '$PLIST_SRC' file..."
+    sudo rm -rf $PLIST_SRC
```
Comment (cpauya) [inline osx/post_installation.sh]: @amodia Why are you removing the .plist here?

File: osx/KA-Lite/KA-Lite/AppDelegate.m [hunk 6]
```diff
@@ -60,7 +60,6 @@ - (void)applicationDidFinishLaunching:(NSNotification *)aNotification {
     // Apply button is clicked, we run `kalite manage setup`.
     bool mustShowPreferences = false;
     @try {
-        checkEnvVars();
```
Comment (cpauya) [inline osx/KA-Lite/KA-Lite/AppDelegate.m]: Why did you remove this `checkEnvVars()` function?

File: osx/KA-Lite/KA-Lite/AppDelegate.m [hunk 7]
```diff
@@ -716,30 +744,26 @@ BOOL setEnvVars() {
...
-    }
     NSString *KaliteHomeStr = [NSString stringWithFormat:@"%@",
                                [NSString stringWithFormat:@"launchctl setenv KALITE_HOME \"%@\"", kaliteHomePath]
                                ];
-    NSString *KalitePythonStr = [NSString stringWithFormat:@"%@",
-                                 [NSString stringWithFormat:@"launchctl setenv KALITE_PYTHON \"%@\"", envKalitePythonStr]
-                                 ];
     
-    NSString *org = @"org.learningequality.kalite";
+    NSString *org = @"org.learningequality.kalite.prefs";
```
Comment (cpauya) [inline osx/KA-Lite/KA-Lite/AppDelegate.m]: Can you add a comment why you named the .plist as such?

File: osx/KA-Lite/KA-Lite/AppDelegate.m [hunk 8]
```diff
@@ -716,30 +745,27 @@ BOOL setEnvVars() {
...
-    }
     NSString *KaliteHomeStr = [NSString stringWithFormat:@"%@",
                                [NSString stringWithFormat:@"launchctl setenv KALITE_HOME \"%@\"", kaliteHomePath]
                                ];
-    NSString *KalitePythonStr = [NSString stringWithFormat:@"%@",
-                                 [NSString stringWithFormat:@"launchctl setenv KALITE_PYTHON \"%@\"", envKalitePythonStr]
-                                 ];
     
-    NSString *org = @"org.learningequality.kalite";
+    // Add prefs to the KALITE_HOME plist name because we have also the same name with KALITE_PYTHON plist at root LaunchAgents.
```
Comment (cpauya) [inline osx/KA-Lite/KA-Lite/AppDelegate.m]: What do you mean by "root LaunchAgents"?  Can you specify the name of the existing .plist for reference?

**Comment by cpauya** [comment]
Ok, tested and merging now.
